waltz-tools/src/main/java/com/wepay/waltz/tools/storage/StorageCli.java

```diff
@@ -780,10 +782,33 @@ private void syncPartitionAssignments(String cliConfigPath) throws Exception {
                     String storageHost = storageConnectString.split(":")[0];
                     int[] partitionIds = assignment.getValue();
                     int storageAdminPort = connectionMetadata.connections.get(storageConnectString);
+                    StorageAdminClient storageAdminClient = null;
+
+                    try {
+                        List<CompletableFuture> futures = new ArrayList<>();
+                        storageAdminClient = openStorageAdminClient(storageHost, storageAdminPort, sslContext,
+                            zkClient, zkRoot);
+
+                        // Get current assigned partitions.
+                        HashMap<Integer, Boolean> assignedPartitionStatus =
+                            (HashMap<Integer, Boolean>) storageAdminClient.getAssignedPartitionStatus().get();
+
+                        for (int partitionId : partitionIds) {
+                            if ((assignedPartitionStatus.get(partitionId) == null) || !assignedPartitionStatus.get(partitionId)) {
+                                futures.add(storageAdminClient.setPartitionAssignment(partitionId, true, false));
+                            }
+                            assignedPartitionStatus.remove(partitionId);
+                        }
+
+                        // Remove the remaining old partitions that are no longer to be handled.
+                        for (int partitionId : assignedPartitionStatus.keySet()) {
+                            storageAdminClient.setPartitionAssignment(partitionId, false, false).get();
```

Comment:
Could we do a *futures.add(storageAdminClient.setPartitionAssignment(partitionId, false, false));* here and wait only once in the *CompletableFuture.allOf.get()*?